deb/publish.go

```diff
@@ -1323,14 +1324,20 @@ func (collection *PublishedRepoCollection) Remove(publishedStorageProvider aptly
 	collection.list[len(collection.list)-1], collection.list[repoPosition], collection.list =
 		nil, collection.list[len(collection.list)-1], collection.list[:len(collection.list)-1]
 
-	if !skipCleanup && len(cleanComponents) > 0 {
-		err = collection.CleanupPrefixComponentFiles(repo.Prefix, cleanComponents,
-			publishedStorageProvider.GetPublishedStorage(storage), collectionFactory, progress)
-		if err != nil {
-			if !force {
-				return fmt.Errorf("cleanup failed, use -force-drop to override: %s", err)
+	if !skipCleanup {
+		publishedStorage := publishedStorageProvider.GetPublishedStorage(storage)
+		if len(cleanComponents) > 0 {
+			err = collection.CleanupPrefixComponentFiles(repo.Prefix, cleanComponents, publishedStorage, collectionFactory, progress)
+			if err != nil {
+				if !force {
+					return fmt.Errorf("cleanup failed, use -force-drop to override: %s", err)
+				}
 			}
 		}
+
+		if localStorage, ok := publishedStorage.(*files.PublishedStorage); ok {
```

Comment:
we need to handle also non local endpoints: s3, azure, swift

Reply:
s3 does not work as local fs - there are no such things as "directories". That's why I opted to `files.PublishedStorage` method and not `aptly.PublishedStorage` interface.

Reply:
Proof link: https://www.reddit.com/r/aws/comments/vizzqr/uploading_empty_folders_as_objects_to_aws_s3/

Reply:
ok, great. what about the others ?

sorry for being picky, I am asking bcs new aptly features should be complete across api/commands/endpoints/databases... just trying to keep things consistent from now on :)

Reply:
Having `func (storage *PublishedStorage) MkDir(_ string) error {` implemented only for local file storage leads me to conclusion that s3, azure and swift do not have such thing as "empty directory"

https://github.com/search?q=repo%3Aaptly-dev%2Faptly+*PublishedStorage%29+MkDir&type=code